File: Fantasy.Base/ServiceModel/ClientRef.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel.Channels;
using System.ServiceModel;

namespace Fantasy.ServiceModel
{
    public class ClientRef<T> : IDisposable
    {

        internal ClientRef(T client)
        {
            this.Client = client;
        }

       

        public void Dispose()
        {

            if (this.State <= CommunicationState.Opened)
            {
                try
                {
                    ((IChannel)this.Client).Close();
                }
                catch
                {
                }
            }
        }


        public CommunicationState State
        {
            get
            {
                return ((IChannel)this.Client).State;
            }
        }
       

        public T Client { get; private set; }


       
    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/ServiceModel/ClientRef.java
package Fantasy.ServiceModel;

public class ClientRef<T> implements IDisposable
{

	public ClientRef(T client)
	{
		this.setClient(client);
	}



	public final void dispose()
	{

		if (this.getState() <= CommunicationState.Opened)
		{
			try
			{
				((IChannel)this.Client).Close();
			}
			catch (java.lang.Exception e)
			{
			}
		}
	}


	public final CommunicationState getState()
	{
		return ((IChannel)this.Client).State;
	}


	private T privateClient;
	public final T getClient()
	{
		return privateClient;
	}
	private void setClient(T value)
	{
		privateClient = value;
	}



}